pkg/logs/config/integration_config.go

```diff
@@ -62,6 +63,13 @@ type LogsConfig struct {
 	// could have been unidirectional but the tailer could not close it in this case.
 	Channel chan *ChannelMessage
 
+	// ChannelTags are the tags attached to messages on Channel; unlike Tags this can be
+	// modified at runtime (as long as ChannelTagsMutex is held).
+	ChannelTags []string
```

Comment:
Can we hide `ChannelTags` and the mutex behind a `Set` and `Get` function on `LogsConfig`? I think it would make this more intuitive and safer to use.

Reply:
Sounds good.  This type could use a major overhaul anyway :)

Reply:
Hm, but that changes
```go
    s.logSource = config.NewLogSource(s.sourceName, &config.LogsConfig{
        Type:        config.StringChannelType,
        Source:      s.source,
        ChannelTags: s.extraTags,
        Channel:     s.logsChan,
    })
```
to
```go
    s.logSource = config.NewLogSource(s.sourceName, &config.LogsConfig{
        Type:        config.StringChannelType,
        Source:      s.source,
        Channel:     s.logsChan,
    })
    s.logSource.SetChannelTags(s.extraTags)
```
this type is _almost_ a pure-data structure, although it has a few random methods.  It's instantiated by parsing JSON, so I think first and foremost it's a data structure.  It already takes a lot of attention to use it correctly (knowing which fields mean what depending on what Type is), so it probably doesn't make sense to "fix" this particular issue in an ad-hoc fashion.

As I think more about how schedulers and launchers handle config, I'm thinking of separating the "parsed JSON" and "configuration for logging" aspects a bit, perhaps using an interface type, so that might be a good time to fix this particular issue.

Reply:
> I'm thinking of separating the "parsed JSON" and "configuration for logging" aspects a bit
+1

ok that's fine.